internal/namespace/oppath.go

```diff
@@ -0,0 +1,52 @@
+package namespace
+
+import (
+	"fmt"
+
+	core "github.com/authzed/spicedb/pkg/proto/core/v1"
+)
+
+/**
+ * decorateRelationOpPaths decorates all SetOperations found within the given relation's rewrite
+ * (if any) with a path indicating each operation's position in the tree of operations.
```

Comment:
> Do we get any benefit from re-using the BDD representations here?

I don't believe so; this is just providing a unique ID for each operation so we can jump into the middle if necessary.

> It might be easy to just ask BDD "are there any negated variables" to determine whether the branch is "conditional"?

Yes, but its not simply if there is one, but whether its reachable.